File: sysdiff.m
function x_dot = f (x, t) 
 
     r = 0.15;
     k = 1.6;
     a = 1.25;
     b = 0.12;
     c = 0.89;
     d = 0.58;
 
     x_dot(1) = r*x(1)*(1 - x(1)/k) - a*x(1)*x(2)/(1 + b*x(1));
     x_dot(2) = c*a*x(1)*x(2)/(1 + b*x(1)) - d*x(2);
  
endfunction

x = lsode ("f", [1; 2], (t = linspace (0, 50, 200)'));

plot (t, x)

File: sys-diff.py
from scipy.integrate import odeint
import numpy as np
import matplotlib.pyplot as plt


# Define a function for the ssoltem of ODEs
def f(y, t):
    y1, y2 = y # unpack variables
    dydt = [y2, -y1] # define derivatives
    return dydt

for n in [0, 0.5, 1, 1.5, 2, 2.5]:
    t = np.linspace(0, 10, 101)
    y0 = [0.0, n]
    sol = odeint(f, y0, t)
    plt.plot(sol[:,0], sol[:,1], 'b-') # path
#   plt.plot([sol[0,0]], [sol[0,1]], 'o') # start
#    plt.plot([sol[-1,0]], [sol[-1,1]], 's') # end

print(sol)
plt.show()
